src/server/check-answers/index.js

```diff
@@ -105,7 +105,16 @@ export class SubmitPageController extends QuestionPageController {
         config.get('featureFlags').biosecurity &&
         application.tasks['biosecurity-map'] !== undefined
       ) {
-        const compressedFile = await this.handleBiosecurityFile(req)
+        const compressedFile = await handleUploadedFile(
+          req,
+          req.yar.get('biosecurity-map')['upload-plan'],
+          this.logger
+        )
+
+        // Error if the file after compression is still too large
+        if (fileSizeInMB(compressedFile.length) > 2) {
```

Comment:
You're doing the comparison at exactly the right place - before base64 encoding - as per the other comment.

Reply:
yes - it's just me re-assuring myself that our checks align with the GDS notify checks.